File: atlas-backend/app/schemas/session.py
"""
Session schemas
"""

from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class SessionCreate(BaseModel):
    """Schema for creating a new study session"""
    task_id: Optional[int] = None
    duration_minutes: int
    start_time: datetime
    notes: Optional[str] = None
    breaks_taken: int = 0
    focus_score: Optional[int] = None  # 1-10 scale


class SessionUpdate(BaseModel):
    """Schema for updating a study session"""
    duration_minutes: Optional[int] = None
    end_time: Optional[datetime] = None
    is_active: Optional[bool] = None
    notes: Optional[str] = None
    breaks_taken: Optional[int] = None
    focus_score: Optional[int] = None


class SessionResponse(BaseModel):
    """Schema for session response"""
    id: int
    task_id: Optional[int] = None
    duration_minutes: int
    start_time: datetime
    end_time: Optional[datetime] = None
    is_active: bool
    notes: Optional[str] = None
    breaks_taken: int
    focus_score: Optional[int] = None
    created_at: datetime
    updated_at: Optional[datetime] = None
    
    class Config:
        from_attributes = True

File: atlas_study_app/lib/core/models/session.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'session.freezed.dart';
part 'session.g.dart';

@freezed
class StudySession with _$StudySession {
  const factory StudySession({
    required int id,
    int? taskId,
    required int durationMinutes,
    required DateTime startTime,
    DateTime? endTime,
    @Default(false) bool isActive,
    String? notes,
    @Default(0) int breaksTaken,
    int? focusScore,
    required DateTime createdAt,
    DateTime? updatedAt,
  }) = _StudySession;

  factory StudySession.fromJson(Map<String, dynamic> json) => _$StudySessionFromJson(json);
}

@freezed
class SessionCreate with _$SessionCreate {
  const factory SessionCreate({
    int? taskId,
    required int durationMinutes,
    required DateTime startTime,
    String? notes,
    @Default(0) int breaksTaken,
    int? focusScore,
  }) = _SessionCreate;

  factory SessionCreate.fromJson(Map<String, dynamic> json) => _$SessionCreateFromJson(json);
}

@freezed
class SessionUpdate with _$SessionUpdate {
  const factory SessionUpdate({
    int? durationMinutes,
    DateTime? endTime,
    bool? isActive,
    String? notes,
    int? breaksTaken,
    int? focusScore,
  }) = _SessionUpdate;

  factory SessionUpdate.fromJson(Map<String, dynamic> json) => _$SessionUpdateFromJson(json);
}
